### PR TITLE
Make it clearer that confirmation is needed for appointments

### LINKED ISSUE #5189: UX 'bug', users dont realise that they need to confirm
### Is your feature request related to a problem? Please describe.

Twice now professional visitors have made appointments on the obscure URL but not confirmed the appointment in the link in the confirming email.  This meant they they thought they had made an appointment and I had no clue.  Two missed appointments!  This has desperately bad optics for me and for Nextcloud.  Initially the visitors thought the appointment system 'worked a treat'.

### Describe the solution you'd like

a) for notifications of unconfirmed appointments.
b) for choice whether confirmation is needed or not (I am happy with the ease and lower security of only using the obscure URL, for now).
c) opportunity for user to manually confirm unconfirmed appointments.
d) for a special email users email address to be used for booking alerts and confirmations.  I can then use the calendar and notifications in more flexible ways.  i.e. someone booking time is one of the highest business priorities and I want to be able to attend to it with care and awareness.
e) to be able to easily provision different copy at each stage - on the url, the confirmation email.

### Describe alternatives you've considered

my kludge is putting this..

For ease and flow.... You *will* need to confirm the appointment for it to appear in my calendar.  Let me write that again.  You absolutely will need to click the link in the email with the subject like, 'Your appointment "Lets Chat" with XXXXXX needs confirmation', before I will have *any* awareness of our meeting.   Please Please do this as I want to meet you!  My apologies for the extra step, which I imagine is designed to protect my calendar from spam.

unfortunately this also appears in the confirmation email.

### Additional context

lessthanthree

### PR DESCRIPTION
**Pull request by pjrobertson**
Does two things to fix #5189

1. Uses an 'email' icon instead of a 'check' icon on the page after submitting an appointment request. The checkmark is confusing and implies that the booking has been confirmed.

Before:
Screen Region 2023-05-15 at 20 49 25

After:
Screen Region 2023-05-15 at 20 51 49

2. Moved the 'confirm' button further up in the confirmation email, so people can see it more clearly that they have to click it
Problem:
Screen Region 2023-05-15 at 20 21 12

I would also say the wording on the page on no.1 should also be improved, but I'm not sure how that works with translations so I've left it for now.

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
Code looks good

File: src/components/Appointments/AppointmentBookingConfirmation.vue [hunk 1]
```diff
@@ -24,15 +24,15 @@
...
+				<EmailFastIcon decorative />
 			</template>
 		</EmptyContent>
 	</div>
 </template>
 
 <script>
 import EmptyContent from '@nextcloud/vue/dist/Components/NcEmptyContent.js'
-import CheckIcon from 'vue-material-design-icons/Check.vue'
+import EmailFastIcon from 'vue-material-design-icons/EmailFast.vue'
```
Comment (tcitworld) [inline src/components/Appointments/AppointmentBookingConfirmation.vue:35]: **issue:** Designers prefer icons from [the official Google Material set](https://fonts.google.com/icons?icon.query=email), not [the MDI project](https://pictogrammers.com/library/mdi/icon/email-fast/).

Comment (pjrobertson) [reply src/components/Appointments/AppointmentBookingConfirmation.vue:35]: Sure, that seems to make sense to me, but I don't know if that's a valid comment for here. A quick search for 'vue-material-design-icons' in the codebase shows 100+ results for this iconset in use. If the material set should be changed, then it should be done all at once:
Screen Region 2023-05-15 at 21 50 17

What this fix is trying to achieve is different. The task of changing icon set is outside the scope of this PR, and should be separate IMO.

Comment (tcitworld) [reply src/components/Appointments/AppointmentBookingConfirmation.vue:35]: Sorry I wasn't clear. I didn't ask to change the icon set, we use that everywhere, I asked for the icon to be picked in the official set of Google Material Icons, not from the community ones from the MDI library. These ones are less stable, [some are removed from time to time](https://pictogrammers.com/docs/library/mdi/releases/changelog/).

Comment (pjrobertson) [reply src/components/Appointments/AppointmentBookingConfirmation.vue:35]: Ah I see, that makes more sense! I have changed the icon to the 'EmailMarkAsUnreadIcon' icon. However, the actual icon is differnet from the link you sent here: https://fonts.google.com/icons?icon.query=email&selected=Material+Symbols+Outlined:outgoing_mail:FILL@0;wght@400;GRAD@0;opsz@48

Also, tbh I don't think it's as clear as the EmailFastIcon, but it should work. Here are some other options:

EmailMarkAsUnreadIcon
Screen Region 2023-05-15 at 22 11 50

EmailIcon
Screen Region 2023-05-15 at 22 11 40

EmailArrowRightOutline
Screen Region 2023-05-15 at 22 11 28

EmailFastIcon
Screen Region 2023-05-15 at 20 51 49

Comment (pjrobertson) [reply src/components/Appointments/AppointmentBookingConfirmation.vue]: Happy to help! I tried to use that icon (mark_email_unread), but it doesn't exist in the 'vue-material-design-icons' project. Here are all the icons that exist: https://www.npmjs.com/package/vue-material-design-icons?activeTab=code

The closest was 'EmailMarkAsUnreadIcon' which if you look at the example above, it doesn't look right/great. Do you have a preferred icon - or how can I get that exact 'mark_email_unread'?

Thanks for the link to the design page - I will definitely take a look! Hopefully I can add a bit more on this app to polish the 'Appointments' feature. It's really powerful but needs just a few small tweaks.

Thanks to you and the team for all the hard work :)

Comment (jancborchardt) [reply src/components/Appointments/AppointmentBookingConfirmation.vue]: @pjrobertson ah ok, strange. Anyway, in that case I think the simple "EmailIcon" is probably best. :)

And yes, looking forward for further contributions from you!

**Review by ChristophWurst — APPROVED** [review]

File: src/components/Appointments/AppointmentBookingConfirmation.vue (hunk 1 above)
Comment (pjrobertson) [reply src/components/Appointments/AppointmentBookingConfirmation.vue]: Alright - I updated to the EmailIcon and pushed the changes. Looking forward to seeing this in the next release. Thanks! :)

**Comment by ChristophWurst** [comment]
```
/home/runner/work/calendar/calendar/src/components/Appointments/AppointmentBookingConfirmation.vue
Error:   35:8  error  'EmailIcon' is defined but never used                       no-unused-vars
Error:   41:3  error  The "CheckIcon" component has been registered but not used  vue/no-unused-components
Error:   41:3  error  'CheckIcon' is not defined                                  no-undef
```

**Comment by pjrobertson** [comment]
Good catch. Updated code and rebased.

**Comment by jancborchardt** [comment]
@pjrobertson if you'd like to contribute more, I can give you a guest account on the instance where our community Talk channels are (such as Design and Frontend). :) Let me know if you are interested, I can add you via your email address.

**Comment by pjrobertson** [comment]
Thanks @jancborchardt  - I'd love to, but time constraints are the main thing stopping me right now. My initial goal is to try and help iron out some bugs to make the appointments feature smooth for my personal/business needs. If I have more time after that, I'll hit you up and ask for a guest account if that's alright!

Thanks :)

**Pull request merged**